File: output/copilot/java/results/max-sum-of-rectangle-no-larger-than-k.java
/** https://leetcode.com/problems/max-sum-of-rectangle-no-larger-than-k/ */
//Given an m x n matrix matrix and an integer k, return the max sum of a rectangle in the matrix such that its sum is no larger than k.

class Solution {
    public int maxSumSubmatrix(int[][] matrix, int k) {
        int m = matrix.length;
        if(m == 0) return 0;
        int n = matrix[0].length;
        if(n == 0) return 0;
        int max = Integer.MIN_VALUE;
        for(int i = 0; i < m; i++) {
            int[] row = matrix[i];
            for(int j = 0; j < n; j++) {
                for(int l = i; l < m; l++) {
                    int[] row2 = matrix[l];
                    int sum = 0;
                    for(int r = j; r < n; r++) {
                        sum += row2[r];
                        if(sum > k) break;
                        if(sum > max) max = sum;
                    }
                }
            }
        }
        return max;
    }
}
    

File: output/copilot/python/results/max-sum-of-rectangle-no-larger-than-k.py
# https://leetcode.com/problems/max-sum-of-rectangle-no-larger-than-k/
#Given an m x n matrix matrix and an integer k, return the max sum of a rectangle in the matrix such that its sum is no larger than k.

class Solution(object):
    def maxSumSubmatrix(self, matrix, k):
        """
        :type matrix: List[List[int]]
        :type k: int
        :rtype: int
        """
        if not matrix or not matrix[0]:
            return 0
        n, m = len(matrix), len(matrix[0])
        max_sum = 0
        for i in range(n):
            for j in range(i, n):
                sum_list = [0 for _ in range(m)]
                for k in range(j + 1):
                    for l in range(m):
                        sum_list[l] += matrix[k][l]
                    sum_max = max(sum_list)
                    if sum_max <= k:
                        max_sum = max(max_sum, sum_max)
        return max_sum
    